pkg/cvo/cvo.go

```diff
@@ -0,0 +1,136 @@
+package cvo
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/openshift/cluster-version-operator/pkg/apis"
+	"github.com/openshift/cluster-version-operator/pkg/apis/clusterversion.openshift.io/v1"
+	"github.com/openshift/cluster-version-operator/pkg/cincinnati"
+	"github.com/openshift/cluster-version-operator/pkg/generated/clientset/versioned"
+	"github.com/openshift/cluster-version-operator/pkg/version"
+
+	"github.com/blang/semver"
+	"github.com/golang/glog"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	// updateCheckPeriod is the amount of time (minus jitter) between update
+	// checks. Jitter will vary between 0% and 100% of updateCheckPeriod.
+	updateCheckPeriod = time.Minute
+
+	namespace = "kube-system"
+
+	customResourceName = "cluster-version-operator"
+)
+
+func StartWorkers(stopCh <-chan struct{}, config *rest.Config, client cincinnati.Client) {
+	go wait.Until(ensureOperatorStatusExists(config), updateCheckPeriod, stopCh)
+	go wait.Until(checkForUpdate(config, client), updateCheckPeriod, stopCh)
+}
+
+func ensureOperatorStatusExists(config *rest.Config) func() {
+	return func() {
+		client, err := clientset.NewForConfig(config)
```

Comment:
move this out of returned func, wait.Until will repeatedly call `ensureOperatorStatusExists`. No use of recreating client everytime.

Reply:
You cannot test this function by using `fakeClient`
eg. https://github.com/openshift/machine-config-operator/blob/e34eeb6f06a3e290f349c0450361f3dafccf6ba8/pkg/controller/render/controller_test.go#L77-L97